site/src/sphinx/client-circuit-breaker.rst

```diff
@@ -1,6 +1,270 @@
+is unresponsive, it will immediately respond with an error and not make a remote calls.
+Please refer to `CircuitBreaker by Martin Fowler`_ and `LINE Engineering blog`_ for more information
+about circuit breaker.
+
+State of ``CircuitBreaker``
+---------------------------
+
+A :api:`CircuitBreaker` can be one of three states which are ``CLOSED``, ``OPEN`` and ``HALF_OPEN``.
+
+- ``CLOSED``
+
+  - Initial state. All requests are treated normally.
+
+- ``OPEN``
+
+  - The state changes to ``OPEN`` once the number of failures divided by the total number of requests exceeds
+    the failure rate. All requests are blocked off responding with :api:`FailFastException`.
+
+- ``HALF_OPEN``.
+
+  - After a certain amount of time in the ``OPEN`` state, it changes to the ``HALF_OPEN`` state which tries
+    to find out if the service is available by sending one request.
+    If the request succeeds, the state changes to ``CLOSED``. If it fails, the state changes to ``OPEN``.
+
+Here is the description how it works:
+
+.. uml::
+
+    @startditaa(--no-separation)
+
+                                       +----------------+
+                                       |                |
+                                       |      OPEN      |
+                                       |                |<-------------------+
+                                       +------------+---+      fail again    |
+                                           ^        |                        |
+                                           |        |                        |
+                                           |        |                        |
+                                           |        |                        |
+     under threshold                       |        |                        |
+         +---+                             |        |                        |
+         |   |                             |        |                        |
+         |   v                             |        |                        |
+    +----+-----------+  exceed threshold   |        |     timeout         +--+-------------+
+    |                +---------------------+        +-------------------->|                |
+    |     CLOSED     |                                                    |    HALF_OPEN   |
+    |                |<---------------------------------------------------+                |
+    +----------------+            back to normal(request succeed)         +----------------+
+
+    @endditaa
+
+``CircuitBreakerClient``
+------------------------
+
+Armeria provides two different :api:`Client` implementations depending on the
+:api:`Request` and :api:`Response` types:
+
+- :api:`CircuitBreakerHttpClient`
+- :api:`CircuitBreakerRpcClient`
+
+Let's use :api:`CircuitBreakerHttpClient` to find out what we can do.
+You can use the decorator_ method in :api:`ClientBuilder` to build a :api:`CircuitBreakerHttpClient`:
```

Comment:
Perhaps we don't need to link to the `client-decorator.html` here? How about just:

```rst
the ``decorator()`` method
```

..?